### DISCUSSION TITLE
Trying to debug a "This Serverless Function has crashed." message

### DISCUSSION THREAD

**Discussion by StokleyO** [vercel/community#3430] [Help]
### Summary

I recently deployed my SvelteKit app to Vercel and have been getting a "This Serverless Function has crashed." message on the front-page. It builds correctly on Vercel, but looks like it may be breaking down when trying to access my internal API call. Code will be in 'steps to reproduce'

### Steps to Reproduce

// @ts-nocheck

import { json, redirect } from "@sveltejs/kit";
import prisma from "../../../lib/prisma";
import bcrypt from 'bcryptjs';

export async function POST(RequestEvent) {
    const { request, cookies } = RequestEvent;
    const formData = await request.formData();
    const username = formData.get('username');
    const password = formData.get('password');

    const userSigningIn = await prisma.user.findUnique({
        where: { username: username }
    })

    const userPasswordCompare = await bcrypt.compare(password, userSigningIn.password)

    if (userPasswordCompare) {
        cookies.set('session', userSigningIn.username, {
            path: '/',
            httpOnly: true,
            sameSite: 'strict',
            maxAge: 60 * 60 * 24
        })
        throw redirect(302, '/')
    } else {
        return json({ body: "Incorrect Credentials" }, { status: 400 })
    }
}

export async function GET(RequestEvent) {
    const { cookies } = RequestEvent;
    const validatedUsername = cookies.get('session');
    
    const userSignedIn = await prisma.user.findUnique({where: {username: validatedUsername}, include: {following: true, favoritedPosts: true, postSentRatings: true, userSentRatings: true}})
    if (userSignedIn == null) {
        return json({isLoggedIn: false})
    }
    return json(userSignedIn, { status: 200 })
}

**Comment by StokleyO** [comment]
These are what my logs look like
![Logs](https://github.com/vercel/community/assets/83097972/5269fa75-1216-4a94-b2a2-dfe439922898)

**Comment by StokleyO** [comment]
Also, if it helps to know, the repo is private on GItHub

**Comment by amyegan** [comment]
If you click an individual row in that log list, it will pull up a side panel with more information. You can also log additional details about the error if you wrap sections of code in [try...catch statements](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch)

**Reply by StokleyO** [reply]
A try/catch is a really good idea! For this app, these are the two logs I keep on getting repeatedly. The weird thing is, I don't even have a favicon. My app.html in my src folder references a png as an icon in my static folder
![Second Log Error](https://github.com/vercel/community/assets/83097972/5c68ea26-506e-4bce-968d-d5c590a0d972)
![First Log Error](https://github.com/vercel/community/assets/83097972/c1392913-d2b0-4cac-b81b-f2162a6f7051)

**Reply by StokleyO** [reply]
Also, thank you for your response!

**Comment by StokleyO** [comment]
Would love to get some help troubleshooting this issue. From what I can tell through reading SvelteKit & Vercel documentation, nothing in my code looks out of the ordinary

**Reply by amyegan** [reply]
If you need help with debugging, please share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so we can work from the same code

**Reply by StokleyO** [reply]
@amyegan I can give you access to the codebase if that works. The Vercel error message is extremely vague so I don't know what in the code is breaking it. The only 3rd party libraries I'm using is Amazon S3 & Prisma. The app has a lot going on but it's fairly boiler plate code, no tricks or flips involved

**Reply by StokleyO** [reply]
Here is my repo @amyegan 
https://github.com/StokleyO/vaulted

**Reply by amyegan** [reply]
Because of the database and S3 connections, I'm not going to be able to run your code to step through the error. But I see there still aren't any [try...catch statements](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch) around the function code. I highly recommend starting with that to log more error info. The added detail should point you in the right direction.

**Reply by StokleyO** [reply]
@amyegan Just added try/catch statements to all my back-end functions. Do you mind taking another look to see if you can help me debug this? I'd love to deploy on Vercel instead of some other service since I've heard so many good things about the product

**Reply by StokleyO** [reply]
Doesn't look like the logs changed at all @amyegan -- My intuition says this is some sort of configuration issue. The site doesn't even load, it's just fully crashed from the beginning and has been since my first deployment. Any help whatsoever here will be a huge life-saver

**Reply by amyegan** [reply]
Sorry, I can't do much to debug this myself without a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) since I can't run the code to step through the error. Next things for you to check:

- Does the deployment work if you skip the API call and use some dummy data instead?
- Are the necessary tokens or environment variables actually available to the API call? Verify that the values actually come through to where the API call is made.

**Reply by StokleyO** [reply]
@amyegan All of the environment variables are correct in Vercel

**Reply by StokleyO** [reply]
@amyegan I don't mean to sound rude, but what exactly does this error mean? This is the part that's confusing me, this doesn't happen when I engage in any action on the page. It doesn't even allow me to open up my site at all. How are any functions/api calls being made at this point? I could understand if it crashed once I interacted with something but it prevents the entire site from opening. Has this issue not been encountered before? I can't express how frustrating this has been. How is there just no solution or explanation for what this error even is? The Vercel docs are just as unhelpful, it doesn't even explain what this error means. Basically just says "try these three things, and if that doesn't work, whelp"
![Serverless Function](https://github.com/vercel/community/assets/83097972/86bab2ad-9d63-48f3-ad00-dc612075fd23)

**Reply by StokleyO** [reply]
@amyegan Tested using dummy data on my front page. Still no dice. Rather than replacing all my API calls with dummy data, I would love if I could just get some context on what this error means exactly so that I can do some deliberate debugging

**Reply by amyegan** [reply]
A [500: FUNCTION_INVOCATION_FAILED](https://vercel.com/docs/errors/application-error-codes#500:-function_invocation_failed) error usually means there was an unhandled exception in the serverless function. You should be able to get more info from the [runtime logs](https://vercel.com/docs/concepts/observability/runtime-logs) or your browser's dev tools. Using [try...catch statements](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch) can help you log additional error details and handle errors in a more informative way.

**Reply by StokleyO** [reply]
@amyegan Thank you. I think I see what the issue is. I was using @ts-ignore to ignore the typescript errors on my back-end API endpoints. I'm adding in type checkers, etc. now

**Reply by StokleyO** [reply]
@amyegan So, I have try catches all over my code. I've fixed every single typescript error that was going on, and I'm still getting the 500 error code. Is there any way you can look through the code base to see if some configuration or anything else stands out at  you? Would love to figure out why this isn't working

**Reply by amyegan** [reply]
If you temporarily remove the API call from the home page and try to just load it as a regular page do you still get the same error?

**Reply by StokleyO** [reply]
@amyegan Yes, I removed the API call entirely and replaced it with dummy data. No fetching whatsoever, and I'm still getting the same "This Serverless Function has crashed" page

**Comment by StokleyO** [comment]
Ended up deploying on Netlify and it took less than 5 minutes. Would love to, and would still, move over to Vercel but for some reason my site will not deploy on there

**Reply by Douglasemmanuel** [reply]
Please could anyone help me out , I got 500 servers less function crashed after deploying Django on vercel

**Comment by ucefkh** [comment]
same issue guys 

![image](https://github.com/vercel/community/assets/1968135/ad9eb1c6-0cd2-447e-8c8e-937b6b14b9db)


Seriously been stuck on this

**Comment by eliawk** [comment]
hello,
trying to deploy my Rermix app on Vercel and it's a nightmare.
The problem is related to every api call, they cause **[Error: Unexpected Server Error]**.
On local everything works fine, just deployed on **Render.com** the same app and it works perfectly.

@amyegan what's the problem with Vercel?

tx

**Reply by amyegan** [reply]
Hey @eliawk. This discussion is about a problem experienced with SvelteKit. Instead of sending unrelated notifications to the everyone else here, I think it would be best if we move this topic to its own discussion. 

Please [start a new discussion](https://github.com/orgs/vercel/discussions/new?category=help) and share as much detail as you can about the project experiencing these errors. If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), that will help us see exactly what you see.

**Reply by eliawk** [reply]
hello @amyegan I've opened another issue here: https://github.com/orgs/vercel/discussions/7159

[a comment by an automated account is not shown]
